benchmarks/polygons/nit/bench_polygon.nit

```diff
@@ -0,0 +1,141 @@
+#This file is part of NIT (http://www.nitlanguage.org).
+#
+# Copyright 2014 Romain Chanoir <romain.chanoir@viacesi.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# a bench for the polygons
+module bench_polygon
+
+intrude import geometry::polygon
+import opts
+
+# Bench adding random vertices
+fun bench_add_vertices(n: Int) do
+	var randompoints = generate_points(n + 1)
+	var points = randompoints.clone
+	points.remove_at(points.length -1)
+	var poly = new ConvexPolygon.with_vertices(randompoints)
+	poly.anti_clock_sort
+	poly.add_vertex(randompoints.last)
+end
+
+# Bench the convex hull algorithm
+fun bench_convex_hull(n: Int) do
+	srand_from(50)
+	var randompoints = new Array[Point[Float]]
+	for i in [0..n] do
+		var point = new Point[Float](300.0.rand, 300.0.rand)
+		randompoints.add(point)
+	end
+	convex_hull(randompoints)
+end
+
+
+# Bench the convexity verificatioon
+fun bench_convexity(n : Int) do
+	var randompoints = generate_points(n)
+	var poly = new ConvexPolygon.with_vertices(randompoints)
+	poly.is_convex
+end
+
+# Bench the point in polygon algorithm
+fun bench_contain(n : Int) do
+	srand_from(50)
+	var randompoints = generate_points(n)
+	var poly = new ConvexPolygon.with_vertices(randompoints)
+	poly.anti_clock_sort
+	var point = new Point[Float](0.0, 0.0)
+	poly.contain(point)
+end
+
+# Bench the sorting of the vertices
+fun bench_sorting(n : Int) do
+	var randompoints = generate_points(n)
+	randompoints.shuffle		
+	var poly = new ConvexPolygon.with_vertices(randompoints)
+	poly.anti_clock_sort
+	
+end
+
+# Bench the intersection test between two polygons
+fun bench_intersection(n : Int) do
+	var r1 = generate_points(n)
+	var r2 = generate_points(n)
+	var poly1 = new ConvexPolygon.with_vertices(r1)
+	var poly2 = new ConvexPolygon.with_vertices(r2)
+	poly1.anti_clock_sort
+	poly2.anti_clock_sort
+	poly1.intersects(poly2)
+end
+
+# Get `n` points from a circle
+fun get_points_on_circle(p: Point[Float], radius: Float, n: Int): Array[Point[Float]] do
+	srand_from(50)
+	var points = new Array[Point[Float]]
+	for i in [0..n] do
+		var angle =  1000.0.rand * pi * 2.0
+		var point = new Point[Float](angle.cos * radius, angle.sin * radius)
+		points.add(point)
+	end
+	return points
+	# var alpha = pi * 2.0 / n.to_f
+	# for i in [0..n] do
+	#	var theta = alpha * i.to_f
+	#	var point = new Point[Float](theta.cos * radius, theta.sin * radius)
+	#	points.add(point)
+	# end
+	# return points
```

Comment:
Commented code, and some parameters are unused `p` and `radius`. You will also get n+1 points using `[0..n]`, you might prefer `n.times`.